### PR TITLE
show /nextreduction

### PR DESCRIPTION
**Pull request by Ekeh**
This creates a page similar to the many bitcoin halving pages out there. This page shows a countdown till the next block reward reduction for decred. It also shows basic information about the block reward itself and links to related documentation. People from our community can link to this page to provide an entry way for new people to be curious about how decred works and learn more.
nextreduction_dcrd

### CODE REVIEW

File: explorer/explorerroutes.go [hunk 1]
```diff
@@ -218,6 +218,69 @@ func (exp *explorerUI) Home(w http.ResponseWriter, r *http.Request) {
...
+	blocks := exp.dataSource.GetExplorerBlocks(int(height), int(height)-8)
+	var bestBlock *types.BlockBasic
+	if blocks == nil {
+		bestBlock = new(types.BlockBasic)
+	} else {
+		bestBlock = blocks[0]
+	}
+	xcBot := exp.xcBot
+	if xcBot != nil {
+		conversions = &homeConversions{
```
Comment (ademuanthony) [inline explorer/explorerroutes.go:247]: `ExchangeRate, StakeDiff and CoinSupply` are not being used.  You may want to create a new struct.

File: explorer/templates.go [hunk 2]
```diff
@@ -398,6 +398,7 @@ func makeTemplateFuncMap(params *chaincfg.Params) template.FuncMap {
 			return dcrutil.Amount(intAmount).ToCoin()
 		},
 		"threeSigFigs": threeSigFigs,
+
```
Comment (ademuanthony) [inline explorer/templates.go:401]: Please remove

File: public/js/controllers/nextreduction_controller.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
+import { Controller } from 'stimulus'
+import globalEventBus from '../services/event_bus_service'
+import humanize from '../helpers/humanize_helper'
+var remainingSeconds, nextReductionBlockHeight
+
+export default class extends Controller {
+  static get targets () {
+    return [
```
Comment (ademuanthony) [inline public/js/controllers/nextreduction_controller.js:8]: Remove unused targets
`difficulty`
Remove duplicates, `powConverted, nextReductionTimer `

File: explorer/templates.go (hunk 2 above)
Comment (Ekeh) [reply explorer/templates.go]: Done.

File: public/js/controllers/nextreduction_controller.js (hunk 3 above)
Comment (Ekeh) [reply public/js/controllers/nextreduction_controller.js:8]: Done.

File: explorer/explorerroutes.go [hunk 4]
```diff
@@ -218,6 +218,69 @@ func (exp *explorerUI) Home(w http.ResponseWriter, r *http.Request) {
...
+		*CommonPageData
+		Info          *types.HomeInfo
+		Conversions   *homeConversions
+		PercentChange float64
+		BestBlock     *types.BlockBasic
+	}{
+		CommonPageData: exp.commonData(r),
+		Info:           homeInfo,
+		Conversions:    conversions,
+		PercentChange:  homeInfo.PoolInfo.PercentTarget - 100,
```
[the remark this answers is not in the document]
Comment (Ekeh) [reply explorer/explorerroutes.go]: Done.

**Comment by chappjc** [comment]
Thanks for the review @ademuanthony .  We'll get this merged asap.

**Comment by xaur** [comment]
This page is a fun idea.

When visualizing that something is about to be reduced, it is nice to show the before and after values, together with absolute and relative (%) changes.

Half-serious: How about adding a text like:

> TL;DR: In ~15 days, DCR issuance will reduce by less than 1%. It is unlikely you will notice anything.
> 
> Sorry to disappoint you, but there are no epic halving events, no supply shocks and no substance for massive news coverage and price speculation surrounding those events. Decred is designed for smooth operation with minimum things to worry about.

**Comment by dnldd** [comment]
The immediate thing that comes to mind is making this a view instead of a page and displaying it on the main dcrdata page. It could be below the `mining` view on the main page and would eliminate the redundant info like POW reward, vote reward etc already displayed on the main page. Thoughts?

File: views/nextreduction.tmpl [hunk 5]
```diff
@@ -0,0 +1,206 @@
+{{define "nextreduction"}}
+{{$conv := .Conversions}}
+<!DOCTYPE html>
+<html lang="en">
+
+{{ template "html-head" "Decred Block Explorer by dcrdata.org"}}
+    {{ template "navbar" . }}
```
Comment (chappjc) [inline views/nextreduction.tmpl:7]: Try to eat white space when appropriate.  e.g. `{{- template "navbar" . -}}`

File: views/nextreduction.tmpl [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+                    </div>
+                     <div class="row mt-1 reduction-block-warpper">
+                        <div class="col-8 col-lg-5 col-md-5 col-sm-8 mb-3 mb-sm-2 mb-md-3 mb-lg-3 reduction-block-inner">
+                            <div class="fs13 text-secondary">
+                                Current Block
+                            </div>
+                            <div class="mono lh1rem fs14-decimal fs24 pt-1 pb-1 d-flex align-items-baseline">
+                                <div class="mtb-0-lr-auto">
+                                 <a class="d-inline-block h3 position-relative" href="/block/{{.BestBlock.Hash}}" data-target="nextreduction.blockHeight">{{.BestBlock.Height}}</a>
+                                <!-- <small style="font-size: 12px;position: relative;top: -30px;left: 62px;">({{.Info.IdxInRewardWindow}})</small> -->
```
Comment (chappjc) [inline views/nextreduction.tmpl:26]: Please remove html comments that are leftovers.

File: explorer/templates.go [hunk 7]
```diff
@@ -600,5 +600,13 @@ func makeTemplateFuncMap(params *chaincfg.Params) template.FuncMap {
 			return intStr
 		},
 		"floor": math.Floor,
+		"getTotalFromRewardWindowRewardWindowSize": func(idx int, max, t int64) int {
+			x := (max - int64(idx)) * t
+			if x == 0 {
+				return 0
+			}
+			allsecs := int(time.Duration(x).Seconds())
+			return allsecs
```
Comment (chappjc) [inline explorer/templates.go:609]: Please document this function, and comments on lines with formulas.

Comment (chappjc) [inline explorer/templates.go:609]: Just `return int(time.Duration...)`

File: explorer/templates.go [hunk 8]
```diff
@@ -600,5 +600,13 @@ func makeTemplateFuncMap(params *chaincfg.Params) template.FuncMap {
 			return intStr
 		},
 		"floor": math.Floor,
+		"getTotalFromRewardWindowRewardWindowSize": func(idx int, max, t int64) int {
+			x := (max - int64(idx)) * t
+			if x == 0 {
+				return 0
+			}
```
Comment (chappjc) [inline explorer/templates.go:607]: The branch seems unnecessary.  It would work as just `return int(time.Duration(x).Seconds())`

File: explorer/templates.go [hunk 9]
```diff
@@ -600,5 +600,13 @@ func makeTemplateFuncMap(params *chaincfg.Params) template.FuncMap {
 			return intStr
 		},
 		"floor": math.Floor,
+		"getTotalFromRewardWindowRewardWindowSize": func(idx int, max, t int64) int {
```
Comment (chappjc) [inline explorer/templates.go:603]: The name "getTotalFromRewardWindowRewardWindowSize" is extremely confusing.  It computes a time/duration, but that's not even suggested by the name.  Plus `RewardWindowRewardWindow`...

In plain language, what does this function do?  It computes the time remaining in the current reward window?
